src/services/user.service.js

```diff
@@ -0,0 +1,47 @@
+import { ApiError } from '../exeptions/api.error.js';
+import { User } from '../models/user.js';
+import { emailService } from '../services/email.service.js';
+import { v4 as uuidv4 } from 'uuid';
+
+function getAllActivated() {
+  return User.findAll({
+    where: {
+      activationToken: null,
+    },
+  });
+}
+
+function normalize({ id, email, name }) {
+  return { id, email, name };
+}
+
+function findByEmail(email) {
+  return User.findOne({ where: { email } });
+}
+
+async function register(name, email, password) {
+  const activationToken = uuidv4(); // token generation
+
+  const existUser = await findByEmail(email);
```

Comment:
Consider handling potential errors from the `findByEmail` function, such as database connection issues, to ensure robust error handling.